ogc-common/src/main/java/io/arlas/server/ogc/common/requestfilter/ElasticFilter.java

```diff
@@ -86,19 +90,36 @@ public static BoolQueryBuilder filter(String constraint, CollectionReferenceDesc
             FilterToElastic filterToElastic = new FilterToElastic(collectionDescription, service);
             try {
                 InputStream stream = new ByteArrayInputStream(constraint.getBytes(StandardCharsets.UTF_8));
-                org.opengis.filter.Filter openGisFilter = (org.opengis.filter.Filter) parser.parse(stream);
-
+                org.opengis.filter.Filter openGisFilter = (org.opengis.filter.Filter)parser.parse(stream);
                 filterToElastic.encode(openGisFilter);
                 ObjectMapper mapper = new ObjectMapper();
                 String filterQuery = mapper.writeValueAsString(filterToElastic.getQueryBuilder());
                 // TODO : find a better way to remove prefix xml in field name
                 boolQuery.filter(QueryBuilders.wrapperQuery(filterQuery.replace("tns:", "")));
             } catch (SAXException e) {
-                throw filterToElastic.ogcException;
+                if (filterToElastic.ogcException != null) {
```

Comment:
I feel like the same code is repeated 3 times, maybe it can be put in a dedicated method?